### DISCUSSION TITLE
`fs.Mount` with `fusermount` fails on Android (termux)

### DISCUSSION THREAD

**Discussion by mseltahir** [rfjakob/gocryptfs#745] [Q&A]
Hello hello,

I'm trying to make `gocryptfs` work on Android (termux). I can build and execute the binary as usual but when I try to decrypt some folder using:
```bash
gocryptfs <encrypted directory> <MOUNTPOINT>
```
It gives me the following error:
```
fs.Mount failed: fusermount exited with code 31
```
I looked up this exit code for `fusermount` but I couldn't find anything useful. And I tried to understand it further by adding this log statement
```go
tlog.Info.Printf("mountpoint: %v\nrootNode: %v", args.mountpoint, rootNode)
```
where the code is failing here: https://github.com/rfjakob/gocryptfs/blob/24b3978715186bed3edc2703e81f165a73c0a74a/mount.go#L475
and I got this output
```
Decrypting master key
mountpoint: /data/data/com.termux/files/home/internal/obsidian
rootNode: i0 ():
fs.Mount failed: fusermount exited with code 31
```
The mount point seem to be correct, but I don't know if `rootNode` is, do you have any insight on how I can go around this? My feeling is that it is a  problem  with the `fusermount` dependency.

Edit:
I'm using the latest version from master ([24b3978](https://github.com/rfjakob/gocryptfs/commit/24b3978715186bed3edc2703e81f165a73c0a74a))

**Comment by rfjakob** [comment]
Hi, not sure about android. Maybe look what
https://github.com/hardcore-sushi/DroidFS does?

**Comment by misterhsp** [comment]
fuse only works as root on Android  And you must use the -nosyslog option.

...

**Comment by mseltahir** [comment]
> fuse only works as root on Android And you must use the -nosyslog option.

Oh that's unfortunate, do you know of any other alternative to this method?

**Comment by okhex** [comment]
`su -M -c /data/data/com.termux/files/usr/bin/gocryptfs -nosyslog -allow_other /sdcard/Efiles /sdcard/goFs/`  this works but there is one problem  it appears on each file operation `Unknown opcode 2016`
